docs/source/en/model_doc/speecht5.mdx

```diff
@@ -0,0 +1,98 @@
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# SpeechT5
+
+## Overview
+
+The SpeechT5 model was proposed in [SpeechT5: Unified-Modal Encoder-Decoder Pre-Training for Spoken Language Processing](https://arxiv.org/abs/2110.07205) by Junyi Ao, Rui Wang, Long Zhou, Chengyi Wang, Shuo Ren, Yu Wu, Shujie Liu, Tom Ko, Qing Li, Yu Zhang, Zhihua Wei, Yao Qian, Jinyu Li, Furu Wei.
+
+The abstract from the paper is the following:
+
+*Motivated by the success of T5 (Text-To-Text Transfer Transformer) in pre-trained natural language processing models, we propose a unified-modal SpeechT5 framework that explores the encoder-decoder pre-training for self-supervised speech/text representation learning. The SpeechT5 framework consists of a shared encoder-decoder network and six modal-specific (speech/text) pre/post-nets. After preprocessing the input speech/text through the pre-nets, the shared encoder-decoder network models the sequence-to-sequence transformation, and then the post-nets generate the output in the speech/text modality based on the output of the decoder. Leveraging large-scale unlabeled speech and text data, we pre-train SpeechT5 to learn a unified-modal representation, hoping to improve the modeling capability for both speech and text. To align the textual and speech information into this unified semantic space, we propose a cross-modal vector quantization approach that randomly mixes up speech/text states with latent units as the interface between encoder and decoder. Extensive evaluations show the superiority of the proposed SpeechT5 framework on a wide variety of spoken language processing tasks, including automatic speech recognition, speech synthesis, speech translation, voice conversion, speech enhancement, and speaker identification.*
+
+This model was contributed by [Matthijs](https://huggingface.co/Matthijs). The original code can be found [here](https://github.com/microsoft/SpeechT5).
+
+## SpeechT5Config
+
+[[autodoc]] SpeechT5Config
+
+## SpeechT5HiFiGANConfig
+
+[[autodoc]] SpeechT5HiFiGANConfig
+
+## SpeechT5Tokenizer
+
+[[autodoc]] SpeechT5Tokenizer
+    - __call__
+    - save_vocabulary
+    - decode
+    - batch_decode
+
+## SpeechT5WaveformFeatureExtractor
+
+[[autodoc]] SpeechT5WaveformFeatureExtractor
+    - __call__
+
+## SpeechT5SpectrogramFeatureExtractor
+
+[[autodoc]] SpeechT5SpectrogramFeatureExtractor
+    - __call__
+
+## SpeechT5ProcessorForSpeechToText
+
+[[autodoc]] SpeechT5ProcessorForSpeechToText
+    - __call__
+    - pad
+    - from_pretrained
+    - save_pretrained
+    - batch_decode
+    - decode
+
+## SpeechT5ProcessorForTextToSpeech
+
+[[autodoc]] SpeechT5ProcessorForTextToSpeech
+    - __call__
+    - from_pretrained
+    - save_pretrained
+
+## SpeechT5ProcessorForSpeechToSpeech
+
+[[autodoc]] SpeechT5ProcessorForSpeechToSpeech
+    - __call__
```

Comment:
Having three different processors means the AutoProcessor API will never work for this model. One processor is enough and you can use the names of the arguments passed to it to make the difference between modalities: `text` and `audio` for input texts and input audio, then `text_target` and `audio_target` for output texts and output audio. This is what is used for tokenizers with text to text models.

Reply:
I don't understand. The AutoProcessor API seems to work fine here? It's 3 different models and each has its own processor.